stl/inc/format

```diff
@@ -343,18 +335,40 @@ public:
     // _Next_arg_id > 0 means automatic
     // _Next_arg_id == -1 means manual
     constexpr size_t next_arg_id() {
-        if (_Next_arg_id >= 0) {
-            return _Next_arg_id++;
+        if (_Next_arg_id == -1) {
+            throw format_error("Can not switch from manual to automatic indexing");
         }
-        throw format_error("Can not switch from manual to automatic indexing");
+
+        return static_cast<size_t>(_Next_arg_id++);
     }
-    constexpr void check_arg_id(size_t _Id) {
-        (void) _Id;
+
+    constexpr void check_arg_id(const size_t _Id) {
+        if (_STD is_constant_evaluated()) {
+            if (_Id >= _Num_args) {
+#ifdef __clang__ // TRANSITION clang-12
+                _TRY_BEGIN
+                throw 42;
+                _CATCH_ALL
+                _CATCH_END
+#else // ^^^ workaround ^^^ / vvv no workaround vvv
+                [[maybe_unused]] const auto _Reinterpret_to_make_not_constexpr = reinterpret_cast<double*>(&_Num_args);
+#endif // !__clang__

```

Comment:
I'm thinking of declaring a non-constexpr function named e.g. `_You_see_this_error_because_arg_id_is_out_of_range`, and calling it here. This should result in a more helpful error message than "subexpression not valid in a constant expression" or "a non-constant (sub-)expression was encountered", as compilers will include the offensive function name in the error message.

Reply:
That was my intend when naming the variable like that. As far as I can tell the variable Name would also appear in the error message.

Reply:
This name explains why the call is not constexpr, but it doesn't explain why we don't want it to be constexpr. IMO the latter is more helpful to users.

Reply:
Absolutely, I should have said directly that your name is mich better